### DISCUSSION TITLE
ELRS will not connect

### DISCUSSION THREAD

**Discussion by Rus1408** [qqqlab/madflight#23] [Q&A]
Good day. Studying your drone code and ran into a problem. Can't connect ELRS radio. Neither factory, removed from a working drone, nor self-made. Development board ESP32-S3 DevKitC1. All sensors work adequately, but the radio cannot be connected. No reactions from the remote control are output to the terminal. The program compiles without errors. I get the same result when flashing test programs for checking the radio. Could you suggest a possible solution?

**Comment by qqqlab** [comment]
Thanks for trying madflight. I use ELRS all the time, and it works fine for me. 

Check that you have ELRS set to CRSF protocol, and 400kbaud. Are the pins connected correctly: tx to rx and rx to tx? Does elrs.lua on the transmitter report that the receiver is connected?

**Reply by Rus1408** [reply]
Yes, the signal level from the receiver is displayed on the transmitter. I also see the receiver version in the settings. When connected to another drone in Betaflight, levels and reactions are displayed normally. Regarding the connection, I tried to change the foam in the firmware and physically reconnect it. No reaction. 

Commented on the terms:
 if(c == CRSF_ADDRESS_FLIGHT_CONTROLLER) Serial.printf("\nreceived: "); Serial.printf("%02x ",c);
and 
Serial.print(" decoded RC: "); for(int i=0;i<16;i++) Serial.printf("%d:%d ",i,rcin_crsf.channel[i]); Serial.println();
In the file "crsf.h"
For monitoring in the terminal. But it's empty again. It seems that the UART port is not initialized to receive data


this is what the terms look like in the terminal regardless of whether the transmitter is turned on and regardless of the position of the sticks and switches:

12:42:43.488 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:43.608 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:43.688 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:43.808 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:43.888 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:44.008 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:44.088 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:44.208 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:44.288 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:44.408 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:44.488 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:44.608 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:44.688 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:44.808 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:44.888 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:45.008 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0	
12:42:45.088 -> rcin_thro:0.38 rcin_roll:+0.00 rcin_pitch:-0.00 rcin_yaw:+0.00 rcin_arm:0 rcin_aux:3 out_armed:0

**Reply by Rus1408** [reply]
It seems to have fixed the error. Serial 1 explicit initialization was added to the setup section.
